metadata/build.rs

```diff
@@ -0,0 +1,10 @@
+fn main() {
+    // Set the host target
+    println!(
+        "cargo:rustc-env=DOCS_RS_METADATA_HOST_TARGET={}",
```

Comment:
For using in a library context I'd prefix the env var with `__` on the astronomical chance there's a conflict and just to communicate that this is a "don't touch" variable

Reply:
I'm pretty sure this environment variable is only set for docs.rs, not for anyone else: https://doc.rust-lang.org/cargo/reference/build-scripts.html#cargorustc-envvarvalue. The `DOCS_RS_METADATA` prefix is only so it doesn't conflict with environment variables you set in your shell, but this will override those anyway.